core/domain/exp_domain.py

```diff
@@ -3113,16 +3115,17 @@ def to_yaml(self) -> str:
             str. The YAML representation of this exploration.
         """
         exp_dict = self.to_dict()
-        # The dictionary returned by `to_dict()` method is ExplorationDict
-        # and ExplorationDict does not contain `schema_version` key, but here
-        # we are defining a `schema_version` key which causes MyPy to throw
-        # error TypedDict has no key 'schema_version'. Thus to silent the error,
-        # we used ignore here.
+        # Here we use MyPy ignore because the dictionary returned by `to_dict()`
+        # method is ExplorationDict and ExplorationDict does not contain
+        # `schema_version` key, but here we are defining a `schema_version` key
+        # which causes MyPy to throw error 'TypedDict has no key schema_version'
+        # thus to silent the error, we used ignore here.
```

Comment:
Just to confirm like how defining a new type will suppress the below error ? `Similarly for the ignore[misc] below` because below we are deleting the key, and the deletion of key from typedict is prohibited .